src/tree-selector/treeSelectorComponent.ts

```diff
@@ -0,0 +1,69 @@
+import * as ng from 'angular';
+
+export class TreeSelectorController {
+
+  public initialData: any;
+  public selectables: any;
+  public openTreeView: any;
+  private reg: any;
+  private url: string;
+  private handleSelected: any;
+
+  /*@ngInject*/
+  constructor(private $http, private $timeout : ng.ITimeoutService){}
+
+  public $onInit() {
+    this.initialData = this.parseSelectable(this.initialData);
+    console.log('selectables: ', this.selectables, this.initialData);
+  }
+
+  private getData = (id?) => {
+    return this.$http({
+      method: 'GET',
+      url: id ? `${this.url}?id=${encodeURIComponent(id)}` : this.url,
+    }).then((response) => {
+      return this.parseSelectable(response.data);
+    });
+  }
+
+  public lazyLoad = (node) => {
+    const dataPromise = this.getData(node.key);
+    return dataPromise.then((data) => {
+      return this.parseSelectable(data);
+    });
+  }
+
+  public parseSelectable = (data: any) => {
+    return data.map((node, key) => {
+      const parsedData = {...node};
+      if(parsedData.nodes) {
+        parsedData.nodes = node.nodes.map((childNode, childKey) => {
+          const child = {...childNode};
+          child.selectable = !!child.key.match(this.selectables);
+          return child;
+        });
+      }
+      parsedData.selectable = parsedData.key.match(this.selectables);
+      return parsedData;
+    });
+  }
+
+  public getSelected = (data) => {
+    this.$timeout(() => this.handleSelected(data));
+    this.openTreeView(false);
+  }
+
+}
+
+export default class TreeSelector implements ng.IComponentOptions {
+  public controller = TreeSelectorController;
+  public controllerAs = 'vm';
+  public template = require('./treeSelector.html');
+  public bindings: any = {
+    initialData: '<',
```

Comment:
I understand the reason why you named this as `initialData`, but in the `treeView` component we already have a `data` and we :heart: consistency. Can these two names unified? (I vote for `data` as it's shorter :wink:)